src/test/java/com/globaroman/bookshopproject/controller/BookControllerTest.java

```diff
@@ -0,0 +1,230 @@
+import com.globaroman.bookshopproject.dto.book.BookSearchParameters;
+import com.globaroman.bookshopproject.dto.book.CreateBookRequestDto;
+import com.globaroman.bookshopproject.model.Book;
+import com.globaroman.bookshopproject.model.Category;
+import com.globaroman.bookshopproject.repository.BookRepository;
+import com.globaroman.bookshopproject.repository.CategoryRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.util.Set;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @BeforeEach
+    public void setUp() {
+        this.mockMvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext).build();
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Create a new book")
+    @Transactional
+    @Rollback
+    void createBook_ValidCreateRequestBookDto() throws Exception {
+        //Given
+        CreateBookRequestDto requestDto = createTestCreateBookRequestDtoTemplate();
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        //When
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/api/books")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isCreated())
+                .andReturn();
+        //Then
+        BookDto actual = objectMapper.readValue(result.getResponse()
+                .getContentAsString(), BookDto.class);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+
+        EqualsBuilder.reflectionEquals(requestDto, actual, "id");
```

Comment:
you dont need ObjectMappet to read value and so on....

spring boot test already has ability to compare expect\actual results via jsonPath()
```suggestion
        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/api/books")
                        .content(jsonRequest)
                        .contentType(MediaType.APPLICATION_JSON))
                .andExpect(status().isCreated())
                .andReturn();
    
```

Reply:
rewrite all tests regarding this comments